pkg/collector/generic_collector.go

```diff
@@ -0,0 +1,33 @@
+package collector
+
+import "sync"
+
+// GenericCollector is a safe collector to collect values
+// and retrieve them later when needed.
+type GenericCollector[T comparable] struct {
```

Comment:
It looks like this is only used in one place (SearchRecent).  Don't think we need this abstraction for now, we can easily append to the slice in the callback.

Reply:
done, inlined the collection.